src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -1078,11 +1079,64 @@ public void Select(int start, int length)
 
 		public void SelectAll() => SelectAllPartial();
 
-
 		partial void SelectPartial(int start, int length);
 
 		partial void SelectAllPartial();
 
+		/// <summary>
+		/// Copies content from the OS clipboard into the text control.
+		/// </summary>
+		public void PasteFromClipboard()
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.High, async () =>
+			{
+				var content = Clipboard.GetContent();
+				var clipboardText = await content.GetTextAsync();
+				var selectionStart = SelectionStart;
+				var selectionLength = SelectionLength;
+				var currentText = Text;
+
+				if (selectionStart >= 0 && selectionLength > 0)
+				{
+					currentText = currentText.Remove(selectionStart, selectionLength);
+				}
+
+				if (selectionStart >= 0)
+				{
+					currentText = currentText.Insert(selectionStart, clipboardText);
+				}
+				else
+				{
+					currentText += clipboardText;
+				}
+
+				Text = currentText;
+			});
+		}
+
+		/// <summary>
+		/// Copies the selected content to the OS clipboard.
+		/// </summary>
+		public void CopySelectionToClipboard()
+		{
+			if (SelectionLength > 0)
+			{
+				var text = SelectedText;
+				var dataPackage = new DataPackage();
+				dataPackage.SetText(text);
+				Clipboard.SetContent(dataPackage);
+			}
+		}
+
+		/// <summary>
+		/// Moves the selected content to the OS clipboard and removes it from the text control.
+		/// </summary>
+		public void CutSelectionToClipboard()
+		{
+			CopySelectionToClipboard();
+			Text = Text.Remove(SelectionStart, SelectionLength);
```

Comment:
Can `SelectionStart` be -1? What happens in this case for Windows and do we have the same behavior here?

Reply:
I don't think it can be -1, mostly because it indicates the caret position, yet it may be interesting to validate this behavior with runtime tests (even if it's only for skia targets?).

Reply:
@jeromelaban I asked because other methods have `selectionStart >= 0` checks

Reply:
Yes, indeed cannot be -1, simplified the code to account for that